src/Sarif.UnitTests/Visitors/MakeUriAbsoluteVisitorTest.cs

```diff
@@ -103,79 +103,117 @@ public void MakeUriAbsoluteVisitor_VisitPhysicalLocation_DoesNotSetUriIfBaseIsNo
         [Fact]
         public void MakeUriAbsoluteVisitor_VisitRun_SetsAbsoluteUriForAllApplicableFiles()
         {
-            Run run = GenerateRunForTest(new Dictionary<string, Uri>()
+            Run run = GenerateRunForTest(new Dictionary<string, FileLocation>()
             {
-                { "%TEST1%", new Uri(@"C:\srcroot\") },
-                { "%TEST2%", new Uri(@"D:\bld\out\") }
+                ["%TEST1%"] = new FileLocation { Uri = new Uri(@"C:\srcroot\") },
+                ["%TEST2%"] = new FileLocation { Uri = new Uri(@"D:\bld\out\") }
             });
-            
+
             MakeUrisAbsoluteVisitor visitor = new MakeUrisAbsoluteVisitor();
 
             var newRun = visitor.VisitRun(run);
+
             // Validate.
-            newRun.Files[0].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/file1.cs");
-            newRun.Files[1].FileLocation.Uri.ToString().Should().Be(@"file://D:/bld/out/src/file2.dll");
-            newRun.Files[2].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip");
-            newRun.Files[3].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip#file3.cs");
-            newRun.Files[4].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip#archive2.gz");
-            newRun.Files[5].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip#archive2.gz/file4.cs");
-            newRun.Files[6].FileLocation.Uri.ToString().Should().Be(@"file://C:/srcroot/src/archive.zip#file5.cs");
+            newRun.Files[0].FileLocation.Uri.ToString().Should().Be(@"file:///C:/srcroot/src/file1.cs");
+            newRun.Files[1].FileLocation.Uri.ToString().Should().Be(@"file:///D:/bld/out/src/file2.dll");
+            newRun.Files[2].FileLocation.Uri.ToString().Should().Be(@"file:///C:/srcroot/src/archive.zip");
+            newRun.Files[3].FileLocation.Uri.ToString().Should().Be(@"file3.cs");
+            newRun.Files[4].FileLocation.Uri.ToString().Should().Be(@"archive2.gz");
+            newRun.Files[5].FileLocation.Uri.ToString().Should().Be(@"file4.cs");
 
             // Operation should zap all uri base ids
-            newRun.Files.Select(f => f.FileLocation.UriBaseId != null).Any().Should().BeFalse();
```

Comment:
>newRun.Files.Select [](start = 12, length = 19)

I messed up using Select here rather than Where on initial update of this test, to elide checks of rewritten files table keys. I was originally confused, thinking Select had been long-lived in our code base. :P mystery solved.

Reply:
Got it.

---
In reply to: [247324741](https://github.com/Microsoft/sarif-sdk/pull/1210#discussion_r247324741) [](ancestors = 247324741)